app/src/org/commcare/android/tasks/ResourceEngineTask.java

```diff
@@ -150,111 +154,138 @@ protected ResourceEngineOutcomes doTaskBackground(String... profileRefs) {
                 ResourceTable recovery = platform.getRecoveryTable();
                 temporary.setStateListener(this);
 
-                /*this populates the upgrade table with resources based on binary files,
-                 * starting with the profile file. If the new profile is not a newer version,
-                 * statgeUpgradeTable doesn't actually pull in all the new references
-                 */
+                // When profileRef points is http, add appropriate dev flags
+                URL profileUrl = null;
+                try {
+                    profileUrl = new URL(profileRef);
+                } catch (MalformedURLException e) {
+                    // profileRef couldn't be parsed as a URL, so don't worry
+                    // about adding dev flags to the url's query
+                }
+
+                // If we want to be using/updating to the latest build of the
+                // app (instead of latest release), add it to the query tags of
+                // the profile reference
+                if (DeveloperPreferences.isNewestAppVersionEnabled() &&
+                        (profileUrl != null) &&
+                        ("https".equals(profileUrl.getProtocol()) ||
+                         "http".equals(profileUrl.getProtocol()))) {
+                    if (profileUrl.getQuery() != null) {
+                        // If the profileRef url already have query strings
+                        // just add a new one to the end
+                        profileRef = profileRef + "&target=build";

```

Comment:
I'd probably write this using the URL tools instead (IE:android URI classes will let you add arguments programatically rather than as strings)